File: backend/src/main/java/com/djuancito/reposteria/entidad/dto/PedidoRequestDTO.java
    package com.djuancito.reposteria.entidad.dto;

    import java.math.BigDecimal;
    import java.util.List;
    import lombok.Data;

  @Data
public class PedidoRequestDTO {
    private Integer usuarioId;
    private String nota;
    private List<DetalleRequestDTO> detalles;
    private BigDecimal subtotal;
    private BigDecimal garantiaPagada;
    private BigDecimal resto;
    private BigDecimal total;
    private String codigoOperacion;  // ← ESTE ES EL CAMPO QUE FALTABA
}

File: frontend/src/app/modelos/pedido-request-dto.ts
// src/app/modelos/pedido-request-dto.ts

export interface PersonalizacionDTO {
  descripcionExtra?: string;
  costoAdicional?: number;
  adicionalesSeleccionados?: Array<{
    adicionalId: number | null;
    nombre: string;
    categoria: string;
    costoAdicional: number;
  }>;
}

export interface DetalleRequestDTO {
  productoId: number | null;
  cantidad: number;
  precioUnitario: number;
  subtotal: number;
  promocionId?: number | null;
  personalizacion?: PersonalizacionDTO;
}

export interface PedidoRequestDTO {
  usuarioId: number;
  nota?: string;
  detalles: DetalleRequestDTO[];
  subtotal: number;
  total: number;
  garantiaPagada: number;
  resto: number;
}
